app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/masterpasswordguidance/MasterPasswordGuidanceScreen.kt

```diff
@@ -0,0 +1,248 @@
+                    Text(
+                        style = MaterialTheme.typography.bodyMedium,
+                        color = MaterialTheme.colorScheme.onSurfaceVariant,
+                        text = stringResource(
+                            R.string.the_longer_your_password_the_more_difficult_to_hack,
+                        ),
+                    )
+                }
+                HorizontalDivider()
+                Column(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(horizontal = 24.dp, vertical = 16.dp),
+                ) {
+                    Text(
+                        text = stringResource(R.string.the_strongest_passwords_are_usually),
+                        style = MaterialTheme.typography.titleSmall,
+                        color = MaterialTheme.colorScheme.onSurface,
+                    )
+                    Spacer(modifier = Modifier.height(8.dp))
+                    BulletTextRow(text = stringResource(R.string.twelve_or_more_characters))
+                    BulletTextRow(
+                        text = stringResource(
+                            R.string.random_and_complex_using_numbers_and_special_characters,
+                        ),
+                    )
+                    BulletTextRow(
+                        text = stringResource(R.string.totally_different_from_your_other_passwords),
+                    )
+                }
+            }
+            Spacer(modifier = Modifier.height(16.dp))
+            TryGeneratorCard(
+                onCardClicked = remember(viewModel) {
+                    {
+                        viewModel.trySendAction(
+                            MasterPasswordGuidanceAction.TryPasswordGeneratorAction,
+                        )
+                    }
+                },
+            )
+            Spacer(modifier = Modifier.navigationBarsPadding())
+        }
+    }
+}
+
+@Composable
+private fun TryGeneratorCard(
+    onCardClicked: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Card(
+        onClick = onCardClicked,
+        shape = RoundedCornerShape(size = 16.dp),
+        modifier = modifier
+            .fillMaxWidth()
+            .wrapContentHeight(),
+        colors = CardDefaults.cardColors(
+            containerColor = MaterialTheme.colorScheme.surfaceContainerLowest,
+        ),
+        elevation = CardDefaults.elevatedCardElevation(),
+    ) {
+        Row(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(16.dp),
+        ) {
+            Icon(
+                painter = rememberVectorPainter(id = R.drawable.ic_generator),
+                contentDescription = null,
+                tint = MaterialTheme.colorScheme.primary,
+            )
+            Spacer(modifier = Modifier.width(16.dp))
+            Column(
+                modifier = Modifier.weight(TWO_THIRDS_WEIGHT),
```

Comment:
Why is this weight needed, Can't you just set specific sizes on the icons?

Reply:
Or even let them be the intrinsic size?

Reply:
without constraining it, the `Text` in the column will consume the rest of the right space and push the last icon out of view.

Reply:
Trying with instrinsic size/ set sizes on the icons gives this 
![image](https://github.com/user-attachments/assets/59d6c3ac-025f-418f-bb90-41757d8f9a22)
which doesn't seem to respect the Spacer between the Column and Icon. Also messing with intrinsic size on the Text or Column gets real wonky.

Reply:
Can we do this instead:

```kotlin
Column(
    modifier = Modifier.weight(weight = 1f, fill = false),
)
```

That way we don't need to deal with the two-thirds thing?

Reply:
most agreeable, making the change post haste!